tests/conftest.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from filelock import FileLock
+from solc_select import solc_select
+
+@pytest.fixture(scope="session")
+def solc_versions_installed():
+    """List of solc versions available in the test environment."""
+    return []
+
+@pytest.fixture(scope="session", autouse=True)
+def register_solc_versions_installed(solc_versions_installed):
+    solc_versions_installed.extend(solc_select.installed_versions())
+
+@pytest.fixture(scope="session")
```

Comment:
Doesn't look like it works well for windows https://github.com/crytic/slither/actions/runs/4545625995/jobs/8013219900?pr=1808

Reply:
I think we could simplify this quite a bit. We can rely on the filesystem instead of carrying a list. What do you think of something like the following instead?

```python
@pytest.fixture(scope="session")
def solc_binary_path():
    def inner(version):
        lock = FileLock(f"{version}.lock")
        with lock.acquire(timeout=60):
            if not solc_select.artifact_path(version).exists()
                print("Installing solc version", version)
                solc_select.install_artifacts([version])
        return solc_select.artifact_path(version)
    return inner
```
and then on the tests you can do

```python
def test_contract_name_collision(solc_binary_path) -> None:
    solc_path = solc_binary_path("0.8.0")
    ...
```